builder/azure/arm/config.go

```diff
@@ -343,7 +343,11 @@ type Config struct {
 	BuildKeyVaultName string `mapstructure:"build_key_vault_name"`
 	// Specify the KeyVault SKU to create during the build. Valid values are
 	// standard or premium. The default value is standard.
-	BuildKeyVaultSKU           string `mapstructure:"build_key_vault_sku"`
+	BuildKeyVaultSKU string `mapstructure:"build_key_vault_sku"`
+
+	// Specify the Disk Encryption Set ID to use to encrypt the OS and data disks created with the VM during the build
+	DiskEncryptionSetId string `mapstructure:"disk_encryption_set_id"`
```

Comment:
I'm not familiar with Azure, but what are the accepted values here? Just looking at the attribute I'm not sure what is expected as far as IDs go for disk encryption

Reply:
Good question!

Azure supports three different types of disk encryption, platform managed keys (the default), customer managed keys (CMKs), or a combination of both keys,
![Screen Shot 2023-01-25 at 1 35 50 PM](https://user-images.githubusercontent.com/23503014/214696889-e8a34eb0-30ad-4dbf-9fea-f013e97c01c8.png)

These CMKs are stored in Disk Encryption Sets, which have IDs,

These IDs are structured like

`/subscriptions/(subscription-uuid)/resourceGroups/jenna/providers/Microsoft.Compute/diskEncryptionSets/name`, 

![Screen Shot 2023-01-25 at 1 27 44 PM](https://user-images.githubusercontent.com/23503014/214697152-61daa0a4-cbab-4306-a08e-604ed2d8a289.png)

It felt simpler to just pass all this in as a string, but we could also take this input in an object that takes in the subscription, resource group, and encryption set of the Disk encryption set, where we assume the subscription and the resource group based on other fields, do you think that would make more sense?

Reply:
Eh, it's mostly my lack of knowledge about Azure expressing itself in my question :)
Both alternatives are fine imho, if the resource ID is clearly exposed in the console, I'd think it's easier to specify rather than specifying the rest. Maybe to help users understand what is expected we could point them to some Azure docs through a link here?

Reply:
I added some clarification about where to find that resource ID in the docs